flocker/node/agents/blockdevice.py

```diff
@@ -1865,6 +2043,16 @@ class DuplicateFilesystemId(Exception):
     """
 
 
+def _blkid(*argv):
+    try:
+        return check_output([b"blkid"] + list(argv))
+    except CalledProcessError as e:
+        if e.returncode == 2:
```

Comment:
man page says
```
If the specified token was not found, or no (specified) devices could be identified, an exit code of 2 is returned.
```

So might it be worth logging and error message including the stderr here in case there's some other reason for this return code.